loadtest/user/userentity/user.go

```diff
@@ -18,7 +17,6 @@ type UserEntity struct {
 	id          int
 	store       store.MutableUserStore
 	client      *model.Client4
-	wsClientMut sync.RWMutex
 	wsClient    *model.WebSocketClient
```

Comment:
Can we also remove this?

Reply:
Yep, that too. I also realized there is one edge case that can be due to this localized state. Previously, we were closing the connection from the Disconnect method, but now we are doing it from the goroutine. But this means that when we close the `closing` channel, it might be that the loop is sleeping during reconnect. And since we are not waiting for the response to come from the `closed` channel, the connection can still remain indefinitely open if it keeps on getting failed to be connected.

It's not the end of the world. But just to be extra-cautious I doubled the timeout on the `closed` channel.

Reply:
I see. What if instead of sleeping we do a `time.After` inside a select that also checks for the closing of the channel?

Reply:
Heh .. that's sneaky :smile: . I think this is getting out of scope for this PR. Mind if I do it in a separate PR ?

Reply:
Sure, let's merge this :+1: